docs/source/examples/mg-mode-part-iii.ipynb

```diff
@@ -0,0 +1,1460 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "With the surfaces defined, we can now construct half-spaces with these surfaces before we use those to create cells"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# Set half-spaces for geometry building\n",
+    "half_spaces = {}\n",
+    "half_spaces['Global'] = \\\n",
+    "    (+surfaces['Global x-'] & -surfaces['Global x+'] &\n",
+    "     +surfaces['Global y-'] & -surfaces['Global y+'])\n",
+    "half_spaces['Assembly'] = \\\n",
+    "    (+surfaces['Assembly x-'] & -surfaces['Assembly x+'] &\n",
+    "     +surfaces['Assembly y-'] & -surfaces['Assembly y+'])\n",
+    "half_spaces['Fuel'] = -surfaces['Fuel Radius']\n",
+    "half_spaces['Clad'] = +surfaces['Fuel Radius'] & -surfaces['Clad Radius']\n",
+    "half_spaces['Pin Moderator'] = +surfaces['Clad Radius']\n",
+    "half_spaces['Top Blade'] = \\\n",
+    "    (+surfaces['Top Blade y-'] & -surfaces['Global y+']) & \\\n",
+    "    (+surfaces['Top Blade x-'] & -surfaces['Top Blade x+'])\n",
+    "half_spaces['Top Steel'] = \\\n",
+    "    (+surfaces['Global x-'] & -surfaces['Top Blade x-']) & \\\n",
+    "    (+surfaces['Top Blade y-'] & -surfaces['Global y+'])\n",
+    "half_spaces['Left Blade'] = \\\n",
+    "    (+surfaces['Left Blade y-'] & -surfaces['Left Blade y+']) & \\\n",
+    "    (+surfaces['Global x-'] & -surfaces['Left Blade x+'])\n",
+    "half_spaces['Left Steel'] = \\\n",
+    "    (+surfaces['Left Blade y+'] & -surfaces['Top Blade y-']) & \\\n",
+    "    (+surfaces['Global x-'] & -surfaces['Left Blade x+'])\n",
+    "half_spaces['Corner Blade'] = \\\n",
+    "    half_spaces['Left Steel'] | half_spaces['Top Steel']\n",
+    "half_spaces['Water Fill'] = \\\n",
+    "    half_spaces['Global'] & ~half_spaces['Assembly'] & \\\n",
+    "    ~half_spaces['Top Blade'] & ~half_spaces['Left Blade'] &\\\n",
+    "    ~half_spaces['Corner Blade']"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We will begin building the 8x8 assembly. To do that we will have to build the cells and universe for each pin type (fuel, gadolinia-fuel, and water)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "universes = {}\n",
+    "cells = {}\n",
+    "\n",
+    "for name, mat, in zip(['Fuel Pin', 'Gd Pin'],\n",
+    "                      [materials['Fuel'], materials['Gad']]):\n",
+    "    universes[name] = openmc.Universe(name=name)\n",
+    "    cells[name] = openmc.Cell(name=name)\n",
+    "    cells[name].fill = mat\n",
+    "    cells[name].region = half_spaces['Fuel']\n",
+    "    universes[name].add_cell(cells[name])\n",
+    "    cells[name + ' Clad'] = openmc.Cell(name=name + ' Clad')\n",
```

Comment:
Could this block include a second nested loop over fuel, clad and water to simplify matters?